typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -384,397 +396,560 @@
 ;; Instances, and Structs (Prefabs?)
 (define/cond-contract (subtype* A t1 t2)
   (-> (listof (cons/c Type? Type?)) Type? Type? (or/c #f list?))
-  (early-return
```

Comment:
Nothing per se... the code got simpler, there were not intermediate bindings, it felt like overkill when `cond` worked just fine and didn't add extra right creep.